app/src/main/java/com/cmput301f20t21/bookfriends/ui/scanner/ScannerAddActivity.java

```diff
@@ -0,0 +1,52 @@
+package com.cmput301f20t21.bookfriends.ui.scanner;
+
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.FragmentContainerView;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+
+import com.cmput301f20t21.bookfriends.R;
+import com.cmput301f20t21.bookfriends.ui.component.ProgressButton;
+import com.cmput301f20t21.bookfriends.ui.profile.ProfileSearchFragment;
+import com.google.android.gms.vision.barcode.Barcode;
+
+public class ScannerAddActivity extends ScannerBaseActivity {
+    FragmentContainerView fragmentContainerView;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        // TODO we might use this fragment to display more book info
+        fragmentContainerView = findViewById(R.id.scanner_extra_fragment_container);
+    }
+
+    @Override
+    protected void onBarcodeReceive(Barcode barcode) {
+        super.onBarcodeReceive(barcode);
+        /* TODO should we show something more?
```

Comment:
The minimum requirement is grabbing the ISBN. If we are just being a complete-the-requirement extremist the current implementation is good enough :), just need someway to pass the information back to the AddEditActivity